File: src/main/java/com/github/aivancioglo/resttest/setters/OAuth1ConsumerKey.java
package com.github.aivancioglo.resttest.setters;

import com.github.aivancioglo.resttest.http.HTTPRequest;

public class OAuth1ConsumerKey<T extends HTTPRequest> implements Setter<T> {
    private String consumerKey;

    public OAuth1ConsumerKey(String consumerKey) {
        this.consumerKey = consumerKey;
    }

    /**
     * Update request.
     * @param request actual request.
     */
    @Override
    public void update(T request) {
        request.setOAuth1ConsumerKey(consumerKey);
    }
}

File: src/main/kotlin/com/github/aivancioglo/resttest/setters/OAuth1ConsumerKey.kt
package com.github.aivancioglo.resttest.setters

import com.github.aivancioglo.resttest.http.HTTPRequest

/**
 * OAuth 1.0 consumer key setter.
 *
 * @constructor is creating variable with OAuth 1.0 consumer key value.
 * @param consumerKey of your requestSpecification.
 */
class OAuth1ConsumerKey<in T : HTTPRequest<*>>(private val consumerKey: String) : Setter<T> {

    /**
     * Setting OAuth 1.0 consumer key to your requestSpecification.
     *
     * @param request that will be updated.
     */
    override fun update(request: T) {
        request.oAuth1.consumerKey = consumerKey
    }
}